http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpDockerDAO.scala

```diff
@@ -0,0 +1,203 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package dao
+
+import cats.effect.Concurrent
+import cats.implicits._
+import cats.mtl.ApplicativeAsk
+import io.chrisdavenport.log4cats.Logger
+import io.circe.Decoder
+import org.broadinstitute.dsde.workbench.leonardo.dao.HttpDockerDAO._
+import org.broadinstitute.dsde.workbench.leonardo.dao.ImageVersion.{Sha, Tag}
+import org.broadinstitute.dsde.workbench.leonardo.model.ClusterTool.{Jupyter, RStudio}
+import org.broadinstitute.dsde.workbench.leonardo.model.ContainerRegistry.{DockerHub, GCR}
+import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterTool, ContainerImage, ContainerRegistry, LeoException}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.http4s._
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.client.middleware.FollowRedirect
+import org.http4s.headers.Authorization
+
+/**
+ * Talks to Docker remote APIs to retrieve manifest information in order to try and figure out
+ * what tool it's running.
+ *
+ * This article was used as a guide: https://ops.tips/blog/inspecting-docker-image-without-pull/
```

Comment:
Again: hope I didn't go too overboard with this but it seems to work :)

If there are any issues in practice we could fall back to having clients explicitly specify images.

Reply:
I wouldn't say 'overboard' but 'fancy'  😊